src/main/java/bio/terra/service/snapshotbuilder/query/QueryBuilder.java

```diff
@@ -0,0 +1,48 @@
+package bio.terra.service.snapshotbuilder.query;
+
+import java.util.List;
+
+public class QueryBuilder {
+  private List<SelectExpression> select;
+  private List<TableVariable> tables;
+  private FilterVariable where;
+  private List<FieldVariable> groupBy;
+  private List<OrderByVariable> orderBy;
+  private Integer limit;
+
+  public QueryBuilder() {}
+
+  public QueryBuilder addSelect(List<SelectExpression> select) {
```

Comment:
I would call this `select()` as `add` doesn't add much here IMO
```suggestion
  public QueryBuilder select(List<SelectExpression> select) {
```